src/core/moves/MetropolisHastingsMove.cpp

```diff
@@ -320,11 +424,34 @@ void MetropolisHastingsMove::performMcmcMove( double prHeat, double lHeat, doubl
         {
             if ( the_node->isClamped() )
             {
-                ln_likelihood_ratio += the_node->getLnProbabilityRatio();
+                try
+                {
+                    ln_likelihood_ratio += the_node->getLnProbabilityRatio();
+                }
+                catch (RbException e)
+                {
+                    ln_likelihood_ratio = RbConstants::Double::neginf;
+
+                    if ( e.getExceptionType() != RbException::MATH_ERROR )
+                    {
+                        throw RbException(e);
+                    }
+                }
             }
             else
             {
-                ln_prior_ratio += the_node->getLnProbabilityRatio();
+                try {
+                    ln_prior_ratio += the_node->getLnProbabilityRatio();
+                }
+                catch (RbException e)
```

Comment:
Catching exception by value.

Reply:
Same as above, what do you by "by value"?

Reply:
Changed.